clang/unittests/Analysis/FlowSensitive/TypeErasedDataflowAnalysisTest.cpp

```diff
@@ -146,6 +146,30 @@ TEST_F(DataflowAnalysisTest, DiagnoseFunctionDiagnoserCalledOnEachElement) {
                                    " (Lifetime ends)\n")));
 }
 
+TEST_F(DataflowAnalysisTest, CanAnalyzeStmt) {
+  std::string Code = R"cc(
+      struct S {int i;};
+      S getAnS() {return S{1};};
+      void foo() {
+        S AnS = getAnS();
+      }
```

Comment:
```suggestion
      struct S { int i; };
      S getAnS() {return S{1};};
      S AnS = getAnS();
```

Reply:
Same reason not to use a global variable as above. Additionally, `AdornedCFG::build` expects to take a `FunctionDecl` in which the provided `Stmt` resides. In the case of `DeclStmt`s, this `FunctionDecl` doesn't seem to actually be used at all, but support may not be guaranteed.

Reply:
I see, thanks for the explanation.